Added an article for replacing the Umbraco Commerce order number generator

eshanrnh opened the pull request:
## Description

Fixes #6904 

## Type of suggestion

* [ ] Typo/grammar fix
* [ ] Updated outdated content
* [ ] New content
* [ ] Updates related to a new version
* [ ] Other

## Product & version (if relevant)
Commverce v15


## Deadline (if relevant)

Anytime

mattbrailsford reviewed (changes requested):
The content is good, but I would have said this was more of a how to guide on how-to-guide than an explanation of the Order Number Generation feature. As such, I'd probably move it to the "How to Guides" section.

sofietoft reviewed (commented):
Adding file names to the code sample would be a big help here 💪 

Does it make sense to add steps? Like, in ordered lists? Doesn't sound like there's a lot of steps, so not sure it's relevant.

eshanrnh commented:
Thanks for the review 🙌

@mattbrailsford I've moved the file to the  "How to Guides" section.
@sofietoft I've added the file name. Since it's just a single step, I haven’t included numbers (ordered list) for the step.

mattbrailsford commented:
I think @sofietoft meant with regards to filenames to add the following tags around code blocks

```
{% code title="CurerrencySwitcher.cshtml" %}

<!-- CODE BLOCK HERE-->

{% endcode %}
```

sofietoft commented:
Ah, yes! Matt is right. That is what I meant 💪 

I think we should add this whenever possible!

mattbrailsford commented on 15/umbraco-commerce/how-to-guides/order-number-customization.md:
````diff
@@ -0,0 +1,50 @@
+
+After creating your custom generator, register it in `Program.cs` to replace the default implementation:
+
+```cs
````
`cs` to `csharp` and needs a code tag for the filename

mattbrailsford commented on 15/umbraco-commerce/how-to-guides/order-number-customization.md:
````diff
@@ -0,0 +1,50 @@
+    }
+}
+```
+{% endcode %}
````
Add a line break between the closing code block and the `code` tag

eshanrnh commented on 15/umbraco-commerce/how-to-guides/order-number-customization.md:
```diff
@@ -0,0 +1,52 @@
+
+To create a custom order number generator, define a class that implements the `IOrderNumberGenerator` interface, for example, `CustomOrderNumberGenerator.cs`:
+
+{% code title="`CustomOrderNumberGenerator.cs`" %}
```
```suggestion
{% code title="CustomOrderNumberGenerator.cs" %}
```

eshanrnh commented:
@mattbrailsford The code block was rendering correctly in Preview mode without the line breaks around it. However, I have now added them based on your suggestion 🙌 I didn't include a code tag for the file name since the rest of the article uses backticks to highlight filenames and methods 🤞

mattbrailsford commented:
@eshanrnh ahhh, excellent. I wasn't sure if the line returns were required or not, but I just knew sometimes they are needed when mixing markdown with other tag types so thought it was best to air on the side of caution.

RE not adding the code tag, I think I still would as that seems to be what we have done in other areas of the docs so even if the sentence leading up to the code block mentions the file name, I think it's still a good idea to add the `{% code %}` block around to show the filename of the code example.

mattbrailsford commented:
LGTM Nice work 👍

eshanrnh commented:
I think the PR should be good to merge now 😀
